ionoscloud/resource_server.go

```diff
@@ -1026,18 +1031,48 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 	return resourceServerRead(ctx, d, meta)
 }
 
+func deleteAllVolumes(ctx context.Context, d *schema.ResourceData, client *ionoscloud.APIClient, server *ionoscloud.Server) diag.Diagnostics {
+	dcId := d.Get("datacenter_id").(string)
+
+	if entities, ok := server.GetEntitiesOk(); ok {
+		if volumes, ok := entities.GetVolumesOk(); ok {
+			if volumes.Items != nil {
+				for _, volume := range *volumes.Items {
+					apiResponse, err := client.VolumesApi.DatacentersVolumesDelete(ctx, dcId, *volume.GetId()).Execute()
+					logApiRequestTime(apiResponse)
+
+					if err != nil {
+						diags := diag.FromErr(fmt.Errorf("error occured while deleting remaining volume %s of server ID %s %w", *server.Properties.BootVolume.Id, d.Id(), err))
+						return diags
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
 func resourceServerDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(SdkBundle).CloudApiClient
 	dcId := d.Get("datacenter_id").(string)
 
-	server, apiResponse, err := client.ServersApi.DatacentersServersFindById(ctx, dcId, d.Id()).Execute()
+	server, apiResponse, err := client.ServersApi.DatacentersServersFindById(ctx, dcId, d.Id()).Depth(2).Execute()
```

Comment:
comment here why we need depth(2)

Reply:
@cristiGuranIonos done in 46795f44335487638eb6bc12084809359ce5a660